lib/components/src/Accordion/AccordionItem.tsx

```diff
@@ -0,0 +1,199 @@
+import { styled } from '@storybook/theming';
+import { nanoid } from 'nanoid';
+import React, { Children, FC, useCallback, useContext, useEffect, useRef, useState } from 'react';
+import { AccordionContext } from './AccordionContext';
+import { AccordionItemContext } from './AccordionItemContext';
+
+// Props are also available from Accordion context provider, but local props
+// takes precedence for scope control
+export type AccordionItemProps = {
+  open?: boolean;
+  indentBody?: boolean;
+  lined?: boolean;
+  narrow?: boolean;
+  preventToggle?: boolean;
+} & React.HTMLAttributes<HTMLLIElement>;
+
+export const AccordionItem: FC<AccordionItemProps> = ({
+  children,
+  open: _open,
+  indentBody: _indentBody,
+  narrow: _narrow,
+  lined: _lined,
+  preventToggle: _preventToggle,
+  ...rest
+}) => {
+  const [open, setOpen] = useState(_open);
+  const context = useContext(AccordionContext);
+  const id = useRef(nanoid());
+  const preventOpen = Children.count(children) < 2;
+  const initialOpen = useRef(_open);
+  const allowDynamicOpen = !preventOpen && _open !== true;
```

Comment:
The same remarks apply on this component. Perhaps I'm missing something important, but to. me:

There should be 1 complex component for state, and a bunch of styling component, that get that state passed down.

Users of the set of components (`AccordionContainer, AccordionItem, `AccordionHeader` etc. MUST ensure they are wrapping these with the state providing component, to make it interact-able.

Reply:
No they don't - they just have to provide the logic themselves and use the controlling props - which is all the Accordion component does, that's the whole idea behind that approach 😃